### PR TITLE
Reorganizing Android Browser versions

### PR DESCRIPTION
**Pull request by afmenez**
- Adding metadata for all versions that had a browser (no results yet);
- Obsoleting ancient versions;
- Removing newer Android versions that are just aliases for Google Chrome (>4);

### CODE REVIEW

**Comment by afmenez** [comment]
This patch addresses #1089 .

**Review by chicoxyzzy — APPROVED** [review]

**Comment by mgol** [comment]
What about Android Browser 4.2 & 4.3?

**Comment by afmenez** [comment]
It's complicated.  As of Android 4.1, Chrome became the default Android browser.  But it is still possible to build the Android Browser unrtil 4.4, and some distros like Cyanogen do.  But it's the same version that was in 4.1.
This link has a decent explanation: https://decadecity.net/blog/2013/11/21/android-browser-versions

**Comment by mgol** [comment]
I don't know if there are any differences between Android Browser in Android 4.2 & 4.3 but Can I Use reports some differences between 4.1 & 4.2/4.3: https://caniuse.com/#compare=android+4.1,android+4.2-4.3

So at least Android 4.2 should still be included.

**Comment by mgol** [comment]
Another issue is that [babel-preset-env](https://www.npmjs.com/package/babel-preset-env) uses `compat-table` data so if you don't include Android 4.2 or 4.3, an entry like `Android >=4.3` will generate a bundle that may not compile in Android 4.3.

I think we should just include every browser version and - in particular - never remove existing ones from the data. Android 4.2 & 4.3 were present so it's best to not remove them.

**Comment by afmenez** [comment]
OK, I have added all the missing versions up to Android 4.4.  I don't see the point of faking versions 5.0 and beyond, since it is just the Chrome browser.

**Comment by mgol** [comment]
Yeah, I don't care that much about versions 4.4 & newer as they're based on Chrome. My remark was relevant for versions up to 4.3.

File: environments.json [hunk 1]
```diff
@@ -1595,50 +1595,117 @@
...
+    "family": "Android Browser",
     "short": "AN 4.3",
     "platformtype": "mobile",
-    "obsolete": true
+    "release": "2013-07-24",
+    "obsolete": "very"
   },
   "android4_4": {
-    "full": "Android Browser",
+    "full": "Android Browser 4.4 (KitKat)",
```
Comment (mgol) [inline environments.json]: I don't think this is correct. Android Browser ended with Android 4.3.

Also, it lacks  the `"equals": "chrome30"` declaration which was previously present. Doesn't this break correctness of the results?

As for me, Android 4.4 & newer was never correct as there's no Android Browser in Android 4.4 & newer so I'd be fine with removing them (If maintainers agree).

**Review by mgol — CHANGES_REQUESTED** [review]

File: environments.json (hunk 1 above)
Comment (afmenez) [reply environments.json]: The info I have is that Chrome became the default browser for Android on JellyBean (4.1), and that's why I had removed 4.1-4.4 initially.
But my info also says that the sources were still available until KitKat (4.4), and some distros shipped with it.  I checked the link that you provided, and it mentions activity on 4.4 too (https://caniuse.com/#compare=android+4,android+4.1,android+4.2-4.3,android+4.4&compare_cats=JS).  And that's why I added them back.

**Comment by afmenez** [comment]
The info I have is that Chrome became the default browser for Android on
JellyBean (4.1), and that's why I had removed 4.1-4.4 initially.
But my info also says that the sources were still available until KitKat
(4.4), and some distros shipped with it.  I checked the link that you
provided, and it mentions activity on 4.4 too (
https://caniuse.com/#compare=android+4,android+4.1,android+4.2-4.3,android+4.4&compare_cats=JS).
And that's why I added them back.

On Tue, Jun 6, 2017 at 7:02 PM, Michał Gołębiowski <notifications@github.com

File: environments.json (hunk 1 above)
Comment (mgol) [reply environments.json]: What Can I Use names as Android 4.4 browser is actually Chrome. You can see they have a separate entry for 4.4.3-4.4.4 as that's when the Chromium-based (!) WebView was updated to a newer Chromium version.

The link you provided at the beginning also says that the WebView in Android 4.4 is Chromium-based and that there is no "Browser" browser there.

Comment (afmenez) [reply environments.json]: Done.

**Comment by chicoxyzzy** [comment]
@mgol do you approve this PR now?

**Review by mgol — APPROVED** [review]
Looks good! Thanks for listening me out. :)

**Pull request merged**